ui/analyse/src/ground.ts

```diff
@@ -6,12 +6,31 @@ import { DrawShape } from 'chessground/draw';
 import resizeHandle from 'common/resize';
 import AnalyseCtrl from './ctrl';
 import * as Prefs from 'common/prefs';
+import { ctrl as makeKeyboardMove } from 'keyboardMove';
 
 export const render = (ctrl: AnalyseCtrl): VNode =>
   h('div.cg-wrap.cgv' + ctrl.cgVersion.js, {
     hook: {
       insert: vnode => {
         ctrl.chessground = site.makeChessground(vnode.elm as HTMLElement, makeConfig(ctrl));
+        if (ctrl.data.pref.keyboardMove) {
+          ctrl.keyboardMove ??= makeKeyboardMove({
+            ...ctrl,
+            data: {
+              ...ctrl.data,
+              // TODO: improve this disgraceful shoehorning
+              player: ctrl.chessground.state.movable as { color: Color },
```

Comment:
This is the questionable bit. We only use this piece of data (`player.color`) in crazyhouse when checking the pockets for a piece, and up until now it could be a static thing. In analysis because color is a dynamic piece of data, we probably want to just refactor a little bit.